File: src/main/java/com/aquasecurity/plugins/trivy/ui/treenodes/IconHelper.java
package com.aquasecurity.plugins.trivy.ui.treenodes;

import com.aquasecurity.plugins.trivy.icons.TrivyIcons;
import com.aquasecurity.plugins.trivy.model.Result;

import javax.swing.*;

public class IconHelper {

    protected static Icon getFileIcon(String type) {
        switch (type) {
            case "cargo":
                return TrivyIcons.Cargo;
            case "cloudformation":
                return TrivyIcons.CloudFormation;
            case "dockerfile":
                return TrivyIcons.Docker;
            case "gemspec":
                return TrivyIcons.Gem;
            case "gobinary":
            case "gomod":
                return TrivyIcons.Go;
            case "jar":
            case "pom":
                return TrivyIcons.Java;
            case "javascript":
                return TrivyIcons.Javascript;
            case "node":
                return TrivyIcons.Node;
            case "npm":
                return TrivyIcons.Npm;
            case "nuget":
                return TrivyIcons.Nuget;
            case "pip":
            case "pipenv":
            case"python-pkg":
                return TrivyIcons.Python;
            case "terraform":
                return TrivyIcons.Terraform;
            case "yarn":
                return TrivyIcons.Yarn;
            case "yaml":
                return TrivyIcons.Yaml;
            default:
                return TrivyIcons.Trivy;
        }
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/ui/treenodes/IconHelper.kt
package com.aquasecurity.plugins.trivy.ui.treenodes

import com.aquasecurity.plugins.trivy.icons.TrivyIcons
import javax.swing.Icon

object IconHelper {
  internal fun getFileIcon(type: String): Icon {
    return when (type) {
      "cargo" -> TrivyIcons.Cargo
      "cloudformation" -> TrivyIcons.CloudFormation
      "dockerfile" -> TrivyIcons.Docker
      "gemspec" -> TrivyIcons.Gem
      "gobinary",
      "gomod" -> TrivyIcons.Go
      "html" -> TrivyIcons.Html
      "jar",
      "pom" -> TrivyIcons.Java
      "javascript" -> TrivyIcons.Javascript
      "js" -> TrivyIcons.Javascript
      "node" -> TrivyIcons.Node
      "npm" -> TrivyIcons.Npm
      "nuget" -> TrivyIcons.Nuget
      "pip",
      "pipenv",
      "python-pkg" -> TrivyIcons.Python
      "secret" -> TrivyIcons.Secret
      "terraform" -> TrivyIcons.Terraform
      "yarn" -> TrivyIcons.Yarn
      "yaml" -> TrivyIcons.Yaml
      else -> TrivyIcons.Text
    }
  }
}
